### PR TITLE
Use 'rbd' to achieve drag and drop functionality

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
RN-153 Sort functionality 

### Changes:
Transfer lists in material ui cannot support drag and drop functionality. Found anothe tool [react-beautiful-dnd](https://www.npmjs.com/package/react-beautiful-dnd) that is very handy to achieve this requirement.

### CODE REVIEW

File: packages/ui-components/src/components/DataLibrary/DataLibrary.js [hunk 1]
```diff
@@ -158,7 +178,7 @@ DataLibrary.defaultProps = {
 
 DataLibrary.propTypes = {
   // options: either an array of options if single data type or map of dataType -> options
-  options: PropTypes.oneOf([PropTypes.object, PropTypes.arrayOf(optionPropType)]).isRequired,
+  options: PropTypes.oneOfType([PropTypes.object, PropTypes.arrayOf(optionPropType)]).isRequired,
```
Comment (billli0) [inline packages/ui-components/src/components/DataLibrary/DataLibrary.js:181]: oneOf is passing an exact value from an array, but not for type. https://jaketrent.com/post/react-oneof-vs-oneoftype

File: packages/ui-components/src/components/DataLibrary/ResultsList.js [hunk 2]
```diff
@@ -44,12 +44,16 @@ export const ResultsList = ({
...
+                  key={option.code}
+                  option={option}
+                  onSelect={onSelect}
+                />
               );
             } else {
               return (
                 <SelectableOption
-                  key={index}
+                  key={option.code}
```
Comment (billli0) [inline packages/ui-components/src/components/DataLibrary/ResultsList.js:56]: When you use index of an array as a key, React will optimize and not render as expected. https://stackoverflow.com/questions/46735483/error-do-not-use-array-index-in-keys

File: packages/ui-components/src/components/DataLibrary/SelectedDataList.js [hunk 3]
```diff
@@ -6,15 +6,21 @@
...
+import { Droppable } from 'react-beautiful-dnd';
 
 const StyledSelectedDataList = styled.div`
   padding-top: 10px;
 `;
 
 export const SelectedDataList = ({ value, onRemove }) => (
-  <StyledSelectedDataList>
-    {value.map((option, index) => {
-      return <SelectedDataCard key={index} option={option} onRemove={onRemove} />;
```
Comment (billli0) [inline packages/ui-components/src/components/DataLibrary/SelectedDataList.js:17]: Same as above.

**Review by IgorNadj — APPROVED** [review]
LGTM

**Pull request merged**